src/structure/widget.ts

```diff
@@ -648,14 +800,87 @@ export class MoleculeViewer {
         }
     }
 
+    /**
+     * Reset the view by re-centering it and zooming to fit the model as much as
+     * possible inside the vies.
+     *
+     * @param center Should we center on the highlighted central atom? If false,
+     *               the camera center/look-at is the center of the whole
+     *               structure.
+     */
     private _resetView(center: boolean): void {
+        // HACK: orthographic camera is broken in 3Dmol
+        // (https://github.com/3dmol/3Dmol.js/issues/434) so we fake on by using
+        // a perspective camera set far away, with a very narrow field of view.
+        this._viewer.setCameraParameters({ fov: 0.01, z: 2000 });
+
         if (center && this._highlighted !== undefined) {
             this._viewer.zoomTo({ serial: this._highlighted.center });
         } else {
             this._viewer.zoomTo();
         }
-
-        this._viewer.zoom(2);
         this._viewer.setSlab(-1000, 1000);
     }
+
+    /**
+     * Rotate the viewed group so that the given direction (in group
+     * coordinates) is aligned with the z axis (in camera space)
+     *
+     * @param direction axis to align with the camera view
+     */
+    private _viewAlong(direction: [number, number, number]): void {
+        const norm = Math.sqrt(
+            direction[0] * direction[0] + direction[1] * direction[1] + direction[2] * direction[2]
+        );
+
+        // angle between Oz and the axis
+        const angle = Math.acos(direction[2] / norm);
+
+        const quaternion = [
+            // rotation axis is direction ^ Oz
+            // rotation[0] * sin(angle / 2)
+            (direction[1] / norm) * Math.sin(angle / 2),
+            // rotation[1] * sin(angle / 2)
+            (-direction[0] / norm) * Math.sin(angle / 2),
+            // rotation[2] * sin(angle / 2)
+            0,
+            Math.cos(angle / 2),
+        ];
```

Comment:
I'm not used to this definition of a quaternion. Shouldn't it be [cos(th/2), n * sin(th/2)]? ala https://mathworld.wolfram.com/Quaternion.html.

Reply:
I used http://euclideanspace.com/maths/geometry/rotations/conversions/angleToQuaternion/index.htm and it work, but I can not say I understand quaternions ... This is the rotation from the Z axis to where we want to put the camera

Reply:
looks like 3Dmol is using a different convention, i.e. `[n * sin(th/2), cos(th/2)]`